src/htmb.gleam

```diff
@@ -25,18 +25,33 @@ import gleam/list
 
 pub type Html
 
+// See https://www.w3.org/TR/2011/WD-html-markup-20110113/syntax.html#void-elements
+const void_elements = [
+  "area", "base", "br", "col", "command", "embed", "hr", "img", "input",
+  "keygen", "link", "meta", "param", "source", "track", "wbr",
+]
+
 pub fn h(
   tag: String,
   attributes: List(#(String, String)),
   children: List(Html),
 ) -> Html {
-  "<"
-  |> string.append(tag)
-  |> list.fold(attributes, _, attribute)
-  |> string.append(">")
-  |> string_builder.from_string
-  |> list.fold(children, _, child)
-  |> string_builder.append("</" <> tag <> ">")
+  let opening =
+    "<"
+    |> string.append(tag)
+    |> list.fold(attributes, _, attribute)
+    |> string.append(">")
+    |> string_builder.from_string
+
+  // Void elements do not have a closing part and cannot accept children
+  // See https://www.w3.org/TR/2011/WD-html-markup-20110113/syntax.html#syntax-elements
+  case list.contains(void_elements, tag) {
```

Comment:
Let's use pattern matching for this rather than seeking through a list, it'll be a little bit faster, and rendering HTML will be in the hot path for request/response

```gleam
case tag {
  "area" | "base" ... -> ...
  _ -> ...
}
```

Reply:
Happy to! Curious what the root cause is in the performance difference? Is this not manifested as a linked list by the compiler?